src/states/node.rs

```diff
@@ -61,8 +62,8 @@ use xor_name::XorName;
 const TICK_TIMEOUT_SECS: u64 = 60;
 /// Time (in seconds) after which a `GetNodeName` request is resent.
 const GET_NODE_NAME_TIMEOUT_SECS: u64 = 60;
-/// Time (in seconds) the members of a section wait for a joining node they sent a network name to.
-const SENT_NETWORK_NAME_TIMEOUT_SECS: u64 = 30;
+/// The number of required leading zero bits for the resource proof
+const RESOURCE_PROOF_DIFFICULTY: u8 = 4;
```

Comment:
It is not just 16 hash attempts required. the node needs to keeps pre-pending till the hash result having 4 leading zero bits.
I tried with 8 and it will fail many of our current tests (taking too much time to resolve the challenge)
also, I think this factor may needs to be refactored when test against the testnet, to figure out what's the most suitable value network can handle